Create several iterations of guessing games.

amymhaddad opened the pull request:
@paulghaddad Please review the guess_word.py file, where the user has to guess a word from the list. My program needs to provide feedback on the guessed word.

amymhaddad committed:
Pick a random word and have the user try to guess it. Provide feedback on the guessed word.

paulghaddad commented on guess_number/guess.py:
```diff
@@ -0,0 +1,71 @@
+import random
+
+random_number = random.randint(0, 100)
```
Nice job using `random`!

paulghaddad commented on guess_number/guess.py:
```diff
@@ -0,0 +1,71 @@
+def guess_number():
+
+    while True:
+        user_guess = int(input("Guess a number: "))
```
Good use of coercion by using `int` here.

paulghaddad commented on guess_number/guess.py:
```diff
@@ -0,0 +1,71 @@
+        if user_guess == random_number:
+            print("Just right!")
+            return random_number
+            break
```
There should be no need for the `break` because the `return` statement on the line above should exit the function.

paulghaddad commented on guess_number/guess.py:
```diff
@@ -0,0 +1,71 @@
+            print("Too high")
+
+
+print(guess_number())
```
You should be able to just call `guess_number()` without the `print` surrounding it because the function itself will print. This is basically redundant.

paulghaddad commented on guess_number/guess.py:
```diff
@@ -0,0 +1,71 @@
+
+
+while True:
+    print(f"Is your guess {str(guess)}?")
```
I believe an f-string will automatically coerce an `int` to a string, so there's probably no need for the `str` call.

paulghaddad commented on guess_number/guess.py:
```diff
@@ -0,0 +1,71 @@
+    elif user_guess > random_number:
+        print("Too high")
+
+    elif user_guess < random_number:
```
Never mine...I reviewed the wrong file 😛

paulghaddad commented on guess_number/guess_word.py:
```diff
@@ -0,0 +1,25 @@
+
+words = ["a", "be", "cat", "dog", "elf"]
+
+random_word = random.choice(words)
```
Great use! 👏

paulghaddad commented on guess_number/guess_word.py:
```diff
@@ -0,0 +1,25 @@
+for word in words:
+    print(word)
+
+print("\n")
```
If you need to print a line break, you can just use `print()`.

paulghaddad commented on guess_number/guess_word.py:
```diff
@@ -0,0 +1,25 @@
+        print("You guessed it!")
+        break
+
+    elif random_word[0] > user_guess[0]:
```
I _believe_ you can just use `random_word > user_guess` because the > should base the comparison on the alphabetized order.

paulghaddad commented on guess_number/guess_word.py:
```diff
@@ -0,0 +1,25 @@
+"""Pick a random word and have the user try to guess it. Provide feedback on the guessed word"""
+import random
+
+words = ["a", "be", "cat", "dog", "elf"]
```
This appears to be a constant so I'd make it uppercase (and choose a more descriptive name).

paulghaddad commented on guess_number/guess_word.py:
```diff
@@ -0,0 +1,25 @@
+import random
+
+words = ["a", "be", "cat", "dog", "elf"]
+
```
Consider wrapping all the work below in one or more well-named functions that you'll call at the end of this file to invoke it.

paulghaddad reviewed (changes requested):
This is mostly good, just some nit-picking. I accidentally reviewed the second file so ignore those comments. I'd like you to refactor the last file here based on my comments and submit a second iteration. Nice work!

amymhaddad commented:
@paulghaddad I updated guess_word.py with these changes:
-Added 2 global variables
-Wrapped all of my work into a function
-Cleaned up my comparisons.

Ready for final review!

amymhaddad commented on guess_number/guess_word.py:
```diff
@@ -0,0 +1,28 @@
+import random
+
+WORD_CHOICES = ["a", "be", "cat", "dog", "elf"]
+RANDOM_WORD = random.choice(WORD_CHOICES)
```
not a constant...always changes because of `choice`.

paulghaddad commented on guess_number/guess_word.py:
```diff
@@ -0,0 +1,28 @@
+        user_guess = input("Guess a word from the list of words: ")
+        if user_guess == RANDOM_WORD:
+            return "You guessed it!"
+
```
Don't believe you need the extra spaces between conditional clauses

paulghaddad reviewed (approved):
This looks good! Just a small style nit-pick. Remember to run the style checker!
